File: cpp/algs/bubble_sort.cpp

#include <cmath>
#include <iostream>
#include <vector>

void bubble_sort(std::vector<int> &array) {
    for (int i = 0; i < array.size(); ++i) {
        for (int j = 0; j < array.size(); ++j) {
            if (array[j] > array[j + 1]) {
                int tmp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = tmp;
            }
        }
    }
}

int main() {
    std::vector<int> array = {0, 3, 2, 1, 4, 5};

    bubble_sort(array);
    for (const auto &item : array) {
        std::cout << item << std::endl;
    }

    return 0;
}

File: python/algs/bubble_sort.py
"""
Сортировка пузырьков -- алгоритм сортировки последовательности чисел.
Описание:
1. Проходим в цикле по массиву n раз где n количество элментов, индекс i до конца массива
2. Если следующий элемент итерации меньше чем текущий, то меняем из местами
Сложность алгоритма O(n*n)
"""


def bubble_sort(arr):
    count = 0  # счетчик для оценки сложности
    for _ in range(0, len(arr)):
        for j in range(0, len(arr) - 1):
            if arr[j] > arr[j + 1]:
                tmp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = tmp
            count += 1
    return arr, count


if __name__ == "__main__":
    arr = [4, 2, 1, 4, 10, 6, 1, 8, 9]
    number = 7
    sorted_arr, count = bubble_sort(arr)
    print(sorted_arr, count)